esp/services/ws_workunits/ws_workunitsQuerySets.cpp

```diff
@@ -1665,8 +1665,18 @@ void copyWorkunitForRecompile(IEspContext &context, IWorkUnitFactory *factory, c
 
     wu->setAction(WUActionCompile);
 
-    SCMStringBuffer token;
-    wu->setSecurityToken(createToken(wuid.str(), context.queryUserId(), context.queryPassword(), token).str());
+    //Digitally sign the workunit
+    if (!wu->setSecuritySignature())
```

Comment:
I think for every place this is called we should evaluate whether we need to update the owner.  For example when copying a workunit perhaps the person copying should be the owner of the copy?  And when we don't wan't to update is it ok to trust creating a signature based on the original owner?

Reply:
@afishbeck Can you comment on @jakesmith suggestion?

Reply:
@RussWhitehead need to mention @afishbeck last in message for him to remain PR pending reviewer.

Reply:
@RussWhitehead yes I think I agree with Jake in that the method should be internal unless you have a compelling need to make it available externally.  If we ever have need to change ownership of a workunit the signature may be updated but that can still be internal.  My comments on the owner of a copied workunit probably being the current user still applies.